app/frontend/src/features/search/SearchPage.tsx

```diff
@@ -159,28 +162,32 @@ export default function SearchPage() {
   useEffect(() => {
     if (!map.current) return;
     const handleMapClickAway = (e: EventData) => {
+      //defaultPrevented is true when a map feature has been clicked
       if (!e.defaultPrevented) {
         handleResultClick(undefined);
       }
     };
 
-    map.current.on("click", handleMapClickAway);
+    //bind event handlers for map events (order matters!)
     map.current.on(
       "click",
       layers.unclusteredPointLayer.id,
       handleMapUserClick
     );
+    map.current.on("click", handleMapClickAway);
```

Comment:
I'm not completely following this... what was the behaviour of the old ordering? It wrongly deselects the user on the map since it calls `handleResultClick` with `undefined`? Whereas now, I'm guessing the `ev.preventDefault` of `handleMapUserClick` would stop it from being called with this new ordering?

Reply:
Steps to reproduce the issue:  (old behavior)
- go to search page
- search for a user
- click 'show user on map button
- the user's map marker is now active on the map
- click this active marker on the map

**expected:** the user's marker should stay active and the map should pan so the marker is centered on the map again if it wasn't centered anymore.
**actual:** the user's marker is deselected

Reply:
without this change, the `handleResultClick` of `SearchPage.tsx` was always called twice when you click a marker on the map. The first time (handleMapClickAway) the value of `user` argument was always undefined. The second time (handleMapUerClick) the value of `user` argument is a User object.

You can test this in production when you add a breakpoint on the first line of `handleResultClick` method.